src/ucp/wireup/wireup_cm.c

```diff
@@ -417,9 +416,10 @@ static void ucp_cm_client_connect_cb(uct_ep_h uct_cm_ep, void *arg,
 err_free_arg:
     ucs_free(progress_arg);
 err_out:
-    ucp_ep->flags &= ~UCP_EP_FLAG_LOCAL_CONNECTED;
+    UCS_ASYNC_BLOCK(&worker->async);
```

Comment:
maybe we need to put lock inside ucp_worker_set_ep_failed?
or at least comment it should be called with lock held?

Reply:
it's commented, in many cases we already have acquired lock

Reply:
where is it commented?

Reply:
https://github.com/openucx/ucx/pull/5166/files#r427865158